src/passes/OptimizeInstructions.cpp

```diff
@@ -458,6 +458,70 @@ struct OptimizeInstructions
           return c;
         }
       }
+      {
+        Const* c;
+        Binary* bin;
+        Expression *x, *y;
+
+        // fneg(x) * fneg(y)   ==>   x * y
+        if (matches(
+              curr,
+              binary(&bin, Mul, unary(Neg, any(&x)), unary(Neg, any(&y))))) {
+          bin->left = x;
+          bin->right = y;
+          return bin;
+        }
+        // fneg(x) / fneg(y)   ==>   x / y
+        if (matches(
+              curr,
+              binary(&bin, DivS, unary(Neg, any(&x)), unary(Neg, any(&y))))) {
+          bin->left = x;
+          bin->right = y;
+          return bin;
+        }
+        // fneg(x * C)   ==>   x * -C
+        // fneg(x / C)   ==>   x / -C
+        // fneg(C / y)   ==>   -C / y
+        if (matches(curr, unary(Neg, binary(&bin, Mul, any(), fval(&c)))) ||
+            matches(curr, unary(Neg, binary(&bin, DivS, any(), fval(&c)))) ||
+            matches(curr, unary(Neg, binary(&bin, DivS, fval(&c), any())))) {
+          c->value = c->value.neg();
+          return bin;
+        }
+        // fneg(x) * y   ==>   fneg(x * y)
+        // x * fneg(y)   ==>   fneg(x * y)
+        // fneg(x) / y   ==>   fneg(x / y)
+        // x / fneg(y)   ==>   fneg(x / y)
+        if ((matches(curr, binary(&bin, Mul, unary(Neg, any(&x)), any(&y))) ||
```

Comment:
These rules do not decrease code size, so a comment should explain the motivation for them.

(I didn't think a lot about this, but it isn't obvious to me why it would be better to canonicalize the `fneg` to the outside?)

Reply:
> These rules do not decrease code size, so a comment should explain the motivation for them.

Yes, it mostly needs for hoist sign for next simplifications. For example:
```ts
(x / -y) * 10.0
-(x / y) * 10.0  // after applied rule `x / fneg(y)`  ==>   `fneg(x / y)`
(x / y) * -10.0  // after applied rule `fneg(x) * C` ==>   `x * -C`
```

Reply:
I added explanation via comment

Reply:
Updated this rules. It's handling differently for now:

`(-x * y) op C` -> `(x * y) op -C`
`(x * -y) op C` -> `(x * y) op -C`
`(-x / y) op C` -> `(x / y) op -C`
`(x / -y) op C` -> `(x / y) op -C`

where `op` is `*`, `/` or `+`